app/models/kaui/tag_definition.rb

```diff
@@ -25,7 +25,7 @@ class TagDefinition < KillBillClient::Model::TagDefinition
 
     ALL_OBJECT_TYPES.each do |object_type|
       define_singleton_method "all_for_#{object_type.downcase}" do |options_for_klient|
-        (all('NONE', options_for_klient).delete_if { |tag_definition| !tag_definition.applicable_object_types.include?(object_type) || tag_definition.is_control_tag }).sort
+        (all('NONE', options_for_klient).delete_if { |tag_definition| !tag_definition.applicable_object_types.include?(object_type || tag_definition.is_control_tag) }).sort
```

Comment:
I understand here we are reverting back to before of Rubucop update to get all tags as before.

but object_type never will be falsey and tag_definition.is_control_tag is never taken into consideration or will it?
`object_type || tag_definition.is_control_tag`
what is the significance of is_control_tag here? or can we remove it from OR condition?

Reply:
- even if tag_definition.is_control_tag is considered (assume object_type is nil or false)
- is_control_tag is a [boolean](https://killbill.github.io/slate/tag-definition.html) value
- tag_definition.applicable_object_types.include?(tag_definition.is_control_tag) will be always false
- as per documention, See [ObjectType](https://github.com/killbill/killbill-api/blob/7ea823b25e8fd6299350ec62d700e265022be34e/src/main/java/org/killbill/billing/ObjectType.java) for a list of the possible values.

so is tag_definition.is_control_tag condition required?

Reply:
Yup, you have a good point.

Actually, before, it was (see https://github.com/killbill/killbill-admin-ui/pull/135/files):

```
!tag_definition.applicable_object_types.include? object_type || tag_definition.is_control_tag
```

which I'm guessing is/was equivalent to:
```
(!tag_definition.applicable_object_types.include?(object_type)) || (tag_definition.is_control_tag)
```

which makes more sense: _return all tag definitions for `object_type` by fetching all tag definitions, excluding (`delete_if`) both the ones which don't apply to that `object_type` and the control tags_.

But it sounds like we want: _return all tag definitions for `object_type` by fetching all tag definitions, excluding (`delete_if`) the ones which don't apply to that `object_type` UNLESS it's a control tag_?

Could we add a few unit tests to make it clearer? Has precedence of operators in Ruby somehow changed in 3.0? 🤔 (we bumped Ruby compatibility version in `master` - have you tried on various Ruby versions?).

Reply:
Good suggestion, here is the test case added to check if this `!tag_definition.applicable_object_types.include?(object_type || tag_definition.is_control_tag)` should alway return all the tags uesrs' or control tags as expected.

Reply:
But to @NarothamSai's point, `object_type || tag_definition.is_control_tag` doesn't make sense?

Reply:
> But to @NarothamSai's point, `object_type || tag_definition.is_control_tag` doesn't make sense?

Do you have any other suggestion to replace this logic with what? As per what i understood reverted the code as it was earlier?

We have also added a test case  to make sure that whatever logic is written here should not break tag_definition functionality and should keep return tag's list.

Reply:
We have removed the check of `tag_definition.is_control_tag` form above logic and added some test case to make sure it should work as expected.